nuxeo-ai-core/src/main/java/org/nuxeo/ai/enrichment/EnrichmentProvider.java

```diff
@@ -56,19 +61,19 @@ public interface EnrichmentProvider {
      */
     @SuppressWarnings("unchecked")
     default RetryPolicy getRetryPolicy() {
```

Comment:
Types should now be given to `RetryPolicy`:
```suggestion
    default RetryPolicy<Collection<AIMetadata>> getRetryPolicy() {
```
Ideally all implementation should receive type too.